### PR TITLE
feat: Count reading sessions in the web reader

### PR DESCRIPTION
**Pull request by chmouel**
Reading in the browser now counts towards statistics. Previously the
web reader only pushed position updates, so a session spent reading
there advanced "Continue reading" but added zero minutes, zero
sessions and no streak day — silently disagreeing with the reader who
had just used it.

The reader now records a session per sitting, bounded by tab
visibility (open when the book becomes visible, closed and sent on
hide/pagehide/unload). Idle time beyond a three-minute cap per gap is
excluded, matching KOReader's page-time accounting; sittings under ten
seconds of active reading are dropped as accidental opens. The
accounting itself lives in a pure, dependency-free module so the rules
are covered by plain unit tests, with a real-Chromium test proving the
wiring end to end.

See ADR-0030 for the full rationale and the edge cases considered
(idle handling, crash loss, relocate-vs-input distinction).

### CODE REVIEW

File: internal/webui/testdata/readersession.test.mjs [hunk 1]
```diff
@@ -0,0 +1,158 @@
+// Unit tests for the reader's session accounting (reader-session.js).
+// Run by TestReaderSessionAccounting through `node --test`; no browser.
+import { test } from 'node:test';
+import assert from 'node:assert/strict';
+import { openSession, IDLE_AFTER_MS, MIN_ACTIVE_MS } from '../static/reader-session.js';
```
[the remark this answers is not in the document]
Comment (chmouel) [reply internal/webui/testdata/readersession.test.mjs:5]: Verified against the fixture: this repo has no `package.json`, but Node's automatic ESM/CJS detection for ambiguous `.js` files (default since Node 20.10+/22) parses `reader-session.js` as ESM from its `export` syntax without any package.json needed. Confirmed locally with Node 22.23.2 (`node --test` passes, 15/15) and it's green in this PR's CI run. Not adding a package.json for this.

File: internal/webui/testdata/readerbrowser.mjs [hunk 2]
```diff
@@ -1170,3 +1179,114 @@ async function nanGuard(evalIn, check) {
...
+  const setHidden = (hidden) => evalIn(`(() => {
+    window.__hidden = ${hidden};
+    document.dispatchEvent(new Event('visibilitychange'));
+    return true;
+  })()`);
+  const wind = (ms) => evalIn(`(() => { window.__skew += ${ms}; return true; })()`);
+  const pushed = async () =>
+    JSON.parse(await evalIn('JSON.stringify(window.__pushedSessions)')).map((body) => {
+      try { return JSON.parse(body).sessions[0]; } catch (e) { return null; }
+    });
```
[the remark this answers is not in the document]
Comment (chmouel) [reply internal/webui/testdata/readerbrowser.mjs:1243]: Good catch — `pushSession` can legitimately batch more than one unsent sitting into a single POST when a close fires while a prior push is still in flight, and the probe's `sessions[0]` would have silently dropped anything past the first. Changed `pushed()` to flatMap every session out of each posted body instead of taking just the first.

File: internal/webui/static/reader-app.js [hunk 3]
```diff
@@ -484,12 +486,105 @@ function schedulePush() {
...
+    const resp = await api("v1/sessions", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      keepalive: true,
+      body: JSON.stringify({ sessions: batch }),
+    });
+    // 2xx: filed. 409: an id already names its sitting. Any other 4xx
+    // is a payload the server will refuse however often it is sent —
+    // an unknown work, say — and replaying cannot help. Only a server
+    // error or no answer at all leaves the batch to try again.
```
[the remark this answers is not in the document]
Comment (chmouel) [reply internal/webui/static/reader-app.js]: You're right, the comment mischaracterized the 409. Per `internal/api/sessions.go`, 409 is `session_id reused with a different payload` — a genuine collision — while re-posting the identical payload is idempotently a 2xx, not a 409. Fixed the comment to say that; the non-retry behavior itself was already correct.
